src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -25,14 +26,16 @@
 
 @register(AlertRule)
 class AlertRuleSerializer(Serializer):
-    def __init__(self, expand=None):
+    def __init__(self, expand: Optional[List[str]] = None):
         self.expand = expand or []
 
-    def get_attrs(self, item_list, user, **kwargs):
+    def get_attrs(
+        self, item_list: Sequence[Any], user: Union[User, RpcUser], **kwargs: Any
+    ) -> DefaultDict[AlertRule, Any]:
         alert_rules = {item.id: item for item in item_list}
         prefetch_related_objects(item_list, "snuba_query__environment")
 
-        result = defaultdict(dict)
+        result = defaultdict(dict)  # type: DefaultDict[AlertRule, Any]
```

Comment:
we should not be using type comments at all -- this should use an annotation `result: dict[AlertRule, dict[ ...]]`

same for any of the other `# type: ...` comments.  I'll put in a rule to block this with a linter since we don't support python 2 at all

Reply:
also the `Any` should be a type here -- `dict[something, something]` but you'll want to figure out the somethings